gcc/rust/typecheck/rust-hir-type-check-expr.cc

```diff
@@ -194,7 +194,14 @@ TypeCheckExpr::visit (HIR::CallExpr &expr)
 	  HirId variant_id;
 	  bool ok = context->lookup_variant_definition (
 	    expr.get_fnexpr ()->get_mappings ().get_hirid (), &variant_id);
-	  rust_assert (ok);
+
+	  if (!ok)
+	    {
+	      rust_error_at (expr.get_locus (),
+			     "expected function, tuple struct or tuple "
+			     "variant, found enum");
```

Comment:
ah, one thing I'd like to see changed before merging this: rustc has an error code for this one, E0423, which we should use as well: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=5845680e152d6b1b32a48c11a70b4d6f

Reply:
Fixed, now output is
`output is /home/liam/gccrs/gcc/testsuite/rust/compile/issue-3046.rs:20:10: error: expected function, tuple struct or tuple variant, found enum [E0423]`

Reply:
perfect, thank you!